internal/provider/edgegw/nat_rule_resource.go

```diff
@@ -386,6 +386,7 @@ func (r *natRuleResource) read(planOrState *NATRuleModel) (stateRefreshed *NATRu
 	}
 	if err != nil {
 		if govcd.ContainsNotFound(err) {
+			diags.AddError("Error NAT Rule Not Found", err.Error())
```

Comment:
This is not necessarily an error. This is why the `found` boolean exists. The error, if it is one, is then returned at the level of the caller of the function read.